Updated Seqr Dockerfile for Python 3.

ShifaSZ opened the pull request:
To upgrade the Python of the Seqr Docker image to Python 3, the below changes have been made:
1. Used `Debian 9(stretch)` instead of `Debian 8(Jessie)`. I tried `Debian 8(Jessie)` but failed because of errors while installing the Google Cloud SDK.
2. Installed Python 3.7 from source code. I tried installing with `apt-get install Python3` but gave up because it can't specify the version of Python 3.
3. Updated the `Postgres` DB tools installation from the `Debian` version change
4. Removed some of the installed packages from the command for installing the Google Cloud tools.

A docker image has been built with the Dockerfile and tested with a local deployment following the guide from https://github.com/macarthur-lab/seqr/blob/master/deploy/LOCAL_INSTALL.md

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -55,9 +64,9 @@ RUN apt-get update \
-RUN echo 'deb http://apt.postgresql.org/pub/repos/apt/ jessie-pgdg main' >> /etc/apt/sources.list.d/postgresql.list \
-    && wget --no-check-certificate -q https://www.postgresql.org/media/keys/ACCC4CF8.asc -O- | apt-key add - \
+# install database clients for debugging (https://www.postgresql.org/download/linux/debian/)
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list \
```
Codacy found an issue: [Avoid additional packages by specifying `--no-install-recommends`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Delete the apt-get lists after installing something](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Either use Wget or Curl but not both](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -17,16 +17,25 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
+    && rm -f Python-3.7.8.tar.xz
+
+# build and install Python
+RUN cd Python-3.7.8 \
```
Codacy found an issue: [Quote this to prevent word splitting.](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Use $(...) notation instead of legacy backticked `...`.](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Use WORKDIR to switch to a directory](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

ShifaSZ commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -71,10 +80,6 @@ RUN CLOUDSDK_CORE_DISABLE_PROMPTS=1 \
     && curl https://sdk.cloud.google.com | bash \
     && apt-get update && apt-get install -y \
         gcc \
-        python-dev \
-        python-setuptools \
-        libffi-dev \
-        libssl-dev \
```
Deleted packages installed above.

hanars commented:
I'm adding @bw2 to this PR since he has worked a lot on creating our docker images. In particular, I'm hoping he might have some insight into 
1) Why Debian needed a major version bump when it has always worked for us in the past
2) How to specify the python version being installed by `apt-get`, or if you were correct and we have to install it from source. I know we use python 3 in the pipeline runner container, so I would ideally like us to be consistent in how we install it
4) Whether the packages you removed were actually there for any reason

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -17,17 +17,27 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     && rm -rf /var/lib/apt/lists/*
 
+# downland and install Python from soruce code
+RUN PYTHON_VERSION="3.7.8" \
```
Codacy found an issue: [Use WORKDIR to switch to a directory](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Quote this to prevent word splitting.](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Quote this to prevent word splitting.](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Use WORKDIR to switch to a directory](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Use WORKDIR to switch to a directory](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Quote this to prevent word splitting.](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -53,37 +63,43 @@ RUN wget https://raw.github.com/miyagawa/cpanminus/master/cpanm -O /usr/bin/cpan
-RUN echo 'deb http://apt.postgresql.org/pub/repos/apt/ jessie-pgdg main' >> /etc/apt/sources.list.d/postgresql.list \
-    && wget --no-check-certificate -q https://www.postgresql.org/media/keys/ACCC4CF8.asc -O- | apt-key add - \
+# install database clients for debugging (https://www.postgresql.org/download/linux/debian/)
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list \
```
Codacy found an issue: [Either use Wget or Curl but not both](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Avoid additional packages by specifying `--no-install-recommends`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Either use Wget or Curl but not both](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Avoid additional packages by specifying `--no-install-recommends`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Either use Wget or Curl but not both](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Avoid additional packages by specifying `--no-install-recommends`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -53,37 +68,43 @@ RUN wget https://raw.github.com/miyagawa/cpanminus/master/cpanm -O /usr/bin/cpan
-RUN echo 'deb http://apt.postgresql.org/pub/repos/apt/ jessie-pgdg main' >> /etc/apt/sources.list.d/postgresql.list \
-    && wget --no-check-certificate -q https://www.postgresql.org/media/keys/ACCC4CF8.asc -O- | apt-key add - \
+# install database clients for debugging (https://www.postgresql.org/download/linux/debian/)
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list \
```
Codacy found an issue: [Avoid additional packages by specifying `--no-install-recommends`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Either use Wget or Curl but not both](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -12,22 +12,37 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
+    && rm -f Python-$PYTHON_VERSION.tar.xz
+
+WORKDIR /Python-$PYTHON_VERSION
+RUN ./configure --enable-optimizations \
```
Codacy found an issue: [Quote this to prevent word splitting.](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -52,38 +67,44 @@ RUN wget https://raw.github.com/miyagawa/cpanminus/master/cpanm -O /usr/bin/cpan
-    && wget --no-check-certificate -q https://www.postgresql.org/media/keys/ACCC4CF8.asc -O- | apt-key add - \
-    && apt-get update && apt-get install -y \
+# install database clients for debugging (https://www.postgresql.org/download/linux/debian/)
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list \
```
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

ShifaSZ commented:
> I'm adding @bw2 to this PR since he has worked a lot on creating our docker images. In particular, I'm hoping he might have some insight into
> 
> 1. Why Debian needed a major version bump when it has always worked for us in the past
> 2. How to specify the python version being installed by `apt-get`, or if you were correct and we have to install it from source. I know we use python 3 in the pipeline runner container, so I would ideally like us to be consistent in how we install it
> 3. Whether the packages you removed were actually there for any reason

ShifaSZ commented:
> I'm adding @bw2 to this PR since he has worked a lot on creating our docker images. In particular, I'm hoping he might have some insight into
> 
> 1. Why Debian needed a major version bump when it has always worked for us in the past
> 2. How to specify the python version being installed by `apt-get`, or if you were correct and we have to install it from source. I know we use python 3 in the pipeline runner container, so I would ideally like us to be consistent in how we install it
> 3. Whether the packages you removed were actually there for any reason

#1. The `libssl-dev` of `Debian:jessie` is Openssl 1.0.0 compatible while `Python 3.7` requires Openssl 1.0.2 or 1.1 compatible. I still can't find a way to solve this issue without upgrading the Debian version.
#2. I tried using `apt-get` to install Python3 on `Debian` but all failed. Installing Python 3 with `apt-get` works well for `ubuntu` but not for `Debian`. Installing Python 3 from source seems the best practice for `Debian`.
#3. The packages I removed are going to install Python 2. I removed them to avoid dual Python versions on the system.

bw2 commented:
This all looks great to me. I agree with Shifa's reasons for upgrading Debian and switching to python install from source, which matches the [pipeline-runner Dockerfile](https://github.com/macarthur-lab/seqr/blob/master/deploy/docker/pipeline-runner/Dockerfile).

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -52,38 +67,44 @@ RUN wget https://raw.github.com/miyagawa/cpanminus/master/cpanm -O /usr/bin/cpan
-    && wget --no-check-certificate -q https://www.postgresql.org/media/keys/ACCC4CF8.asc -O- | apt-key add - \
-    && apt-get update && apt-get install -y \
+# install database clients for debugging (https://www.postgresql.org/download/linux/debian/)
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list \
```
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
pinning a version caused this to fail, so reverted back

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented on deploy/docker/seqr/Dockerfile:
```diff
@@ -52,38 +68,44 @@ RUN wget https://raw.github.com/miyagawa/cpanminus/master/cpanm -O /usr/bin/cpan
-        postgresql-client \
-        libpq-dev
+# install database clients for debugging (https://www.postgresql.org/download/linux/debian/)
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list \
```
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars replied:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

ShifaSZ commented on seqr/views/apis/project_api.py:
```diff
@@ -306,7 +306,7 @@ def _get_json_for_variant_tag_types(project):
 
     project_variant_tags.append(note_tag_type)
-    project_variant_tags = sorted(project_variant_tags, key=lambda variant_tag_type: variant_tag_type['order'])
+    project_variant_tags = sorted(project_variant_tags, key=lambda variant_tag_type: 0 if variant_tag_type['order'] is None else variant_tag_type['order'])
```
Without this change, there is an error: `TypeError: '<' not supported between instances of 'NoneType' and 'float'` in Python 3.

hanars replied:
1) `order` isn't ever `None` so this should be impossible to hit, which concerns me
2) If you really need to change this `variant_tag_type['order'] or 0` is neater

ShifaSZ replied:
There is one of the variant_tag_type's which variant_tag_type['order'] is None. We should either fix the data in the database or update the code.

hanars replied:
Okay, update the code to `variant_tag_type['order'] or 0`

ShifaSZ commented on deploy/kubernetes/gcloud-dev-settings.yaml:
```diff
@@ -15,7 +15,7 @@ ELASTICSEARCH_SERVICE_EXTERNAL_PORT: 9200
 GCLOUD_ZONE: 'us-central1-b'
 
-SEQR_GIT_BRANCH: 'master'
+SEQR_GIT_BRANCH: 'py3-seqr-docker'
```
Temporary change to use branch `py3-seqr-docker`

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented on ui/dist/asset-manifest.json:
```diff
@@ -1,8 +1,8 @@
-  "app.js": "/static/app-65d69047.js",
-  "app.css.map": "/static/app.d42031fb.css.map",
-  "app.js.map": "/static/app-65d69047.js.map",
+  "app.css": "/static/app.10bd0b1e.css",
```
why are the static assets different from dev? That looks like a merge issue to me and should be reset

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity increasing per file
==============================
- seqr/views/apis/project_api.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)

hanars commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5815111&bid=19037649)
